compiler/optimizer/GlobalRegisterAllocator.cpp

```diff
@@ -3961,9 +3961,9 @@ TR_GlobalRegisterAllocator::markAutosUsedIn(
          }
 
       if (!canSkip &&
-          ((node->getOpCodeValue() == TR::iadd) || (node->getOpCodeValue() == TR::isub) ||
-           (node->getOpCodeValue() == TR::iand) || (node->getOpCodeValue() == TR::iloadi) || (node->getOpCodeValue() == TR::ixor)) ||
-          ((node->getOpCodeValue() == TR::iload) /* && !node->getSymbolReference()->getSymbol()->isAutoOrParm() */))
+          (((node->getOpCodeValue() == TR::iadd) || (node->getOpCodeValue() == TR::isub) ||
+            (node->getOpCodeValue() == TR::iand) || (node->getOpCodeValue() == TR::iloadi) || (node->getOpCodeValue() == TR::ixor)) ||
+           ((node->getOpCodeValue() == TR::iload) /* && !node->getSymbolReference()->getSymbol()->isAutoOrParm() */)))
```

Comment:
If I'm matching these parentheses correctly, the new code is not equivalent to the old one because the `!canSkip &&` did not apply to the `(node->getOpCodeValue() == TR::iload)`. Looking through the rest of the code quickly, I _think_ the fix is correct. I'd like confirmation that the change was intentional, since the commit message says "These changes fix no bugs, just eliminate some warnings".

Reply:
it was intentional (I forgot about it when I updated the comment <blush>).

@andrewcraik @vijaysun would you please confirm that you agree with this change, please?

Reply:
It is not equivalent in that canSkip did not apply to TR::iload and yet canSkip is initialized to false and the entire if that could set it to true is off due to `0 &&`. From a functional perspective right now this should be fine. @vijaysun-omr may have better context to know if there is a potential problem, but it does look ok to me in the context.

Reply:
ok, had a discussion with @andrewcraik and @vijaysun on this code change. It turns out the change is actually irrelevant due to other disabled code blocks (so `canSkip` is actually false for this if condition). There are other significant blocks of code in this function that have been effectively disabled in some distant past investigation and nobody knows of any plan to re-enable. Vijay has promised to take a gander through that code to see if he believes there is value in keeping any of it. If not, then I'll adapt my commit to just rip out the dead code to simplify the function, which should have the side benefit of also eliminating the warning :) .

Reply:
I think the main piece of sign extension code that I know was used (and still seems to be enabled) is at    

``` if (!enableNewSignExtGRA && /* !hasCatchBlock && */ node->getOpCode().isLoadVarDirect() && ```

The snippet of code under discussion here is from some prototype that was never enabled (I don't remember details now) but I think it's safe to delete the code for "enableNewSignExtGRA" being true in my opinion because it was never enabled and given that it's several years since we last touched this area of the codebase, I don't think it needs to clutter the codebase.